src/ucp/wireup/select.c

```diff
@@ -1176,14 +1201,44 @@ ucp_wireup_add_rma_bw_lanes(const ucp_wireup_select_params_t *select_params,
     bw_info.usage             = UCP_WIREUP_LANE_USAGE_RMA_BW;
 
     UCS_STATIC_ASSERT(UCS_MEMORY_TYPE_HOST == 0);
-    for (mem_type = UCS_MEMORY_TYPE_HOST; mem_type < UCS_MEMORY_TYPE_LAST; mem_type++) {
-        if (!context->mem_type_access_tls[mem_type]) {
-            continue;
+    /* First checked RNDV mode has to be a mode specified in config */
+    ucs_assert(rndv_modes[0] == context->config.ext.rndv_mode);
+
+    /* RNDV protocol can't mix different schemes, i.e. wireup has to
+     * select lanes with the same iface flags depends on a requested
+     * RNDV scheme. If no lanes were selected in the loop below, try
+     * other schemes, if RNDV scheme requested by user is auto */
+    for (i = 0; i < ucs_array_size(rndv_modes); i++) {
```

Comment:
can't we just add lanes according to the configured RNDV mode? why need to iterate over all modes?

Reply:
we do this - we check below for `(context->config.ext.rndv_mode != UCP_RNDV_MODE_AUTO)`, we do no looping more if this is not AUTO
this is loop is needed if there are no found UCTs that support `GET | PUT` when `rdnv_mode = AUTO` and then we check GET and then PUT

Reply:
need to document this in 1207..1210, it's not clear..

Reply:
> need to document this in 1207..1210, it's not clear..

done